physics/boyles_law.py

```diff
@@ -0,0 +1,181 @@
+"""
+Title : Implementation of Boyle's law.
+
+Description :
+    Boyle's law, also referred to as the Boyle-Mariotte law, or Mariotte's law
+    (especially in France), is an experimental gas law that describes the relationship
+    between pressure and volume of a confined gas.
+
+    Boyle's law is a gas law which states that the pressure exerted by a gas
+    (of a given mass, kept at a constant temperature) is inversely proportional to the
+    volume occupied by it.
+
+    In other words, the pressure and volume of a gas are inversely proportional to each
+    other as long as the temperature and the quantity of gas are kept constant.
+    Boyle's law was put forward by the Anglo-Irish chemist Robert Boyle in the year 1662
+
+    For a gas, the relationship between volume and pressure (at constant mass and
+    temperature) can be expressed mathematically as follows.
+
+    P ∝ (1/V)
+
+    Where P is the pressure exerted by the gas and V is the volume occupied by it. This
+    proportionality can be converted into an equation by adding a constant, k.
+
+    P = k*(1/V) ⇒ PV = k
+
+    Boyle's law states that when the temperature of a given mass of confined gas is
+    constant,the product of its pressure and volume is also constant. When comparing the
+    same substance under two different sets of conditions, the law can be expressed as:
+
+    P1V1 = P2V2
+
+    Where,
+
+    P1 is the initial pressure exerted by the gas in Pascals (P)
+    V1 is the initial volume occupied by the gas Litres (L)
+    P2 is the final pressure exerted by the gas Pascals (P)
+    V2 is the final volume occupied by the gas Litres (L)
+
+    This equation can be used to predict the increase in the pressure exerted by a gas
+    on the walls of its container when the volume of its container is decreased
+    (and its quantity and absolute temperature remain unchanged).
+
+Sources :
+    https://en.wikipedia.org/wiki/Boyle%27s_law
+    https://byjus.com/chemistry/boyles-law/
+"""
+
+valid_variables: list[str] = ["v1", "v2", "p1", "p2"]
+
+
+def check_validity(values: dict[str, float]) -> bool:
+    """
+
+    Function takes dictionary as an input and returns True if the input
+    is valid
+
+    >>> check_validity({})
+    Traceback (most recent call last):
+        ...
+    ValueError: Invalid input expected 3 items got 0
+
+    >>> check_validity({'v1':2,'v2':4,'k':6})
+    Traceback (most recent call last):
+        ...
+    ValueError: Invalid input k is not a valid variable
+
+    >>> check_validity({'v1':2,'v2':4,'p1':6})
+    True
+
+    """
+    if len(values) == 3:
+        for value in values:
+            if value not in valid_variables:
+                msg = f"Invalid input {value} is not a valid variable"
+                raise ValueError(msg)
+        return True
+    else:
+        msg = f"Invalid input expected {3} items got {len(values)}"
+        raise ValueError(msg)
```

Comment:
Sure I will not return anything if input is valid.

I have changed as you have suggested in 72 to 74 in above snippet
I was getting 'NameError: name 'value' is not defined' in line 77 of above snippet. So I have kept simple if inside for instead of any